pkg/sdn/api/netid.go

```diff
@@ -0,0 +1,89 @@
+package api
+
+// Accessor methods to annotate NetNamespace for multitenant support
+import (
+	"fmt"
+	"strings"
+)
+
+type PodNetworkAction string
+
+const (
+	// Maximum VXLAN Virtual Network Identifier(VNID) as per RFC#7348
+	MaxVNID = uint32((1 << 24) - 1)
+	// VNID: 1 to 9 are internally reserved for any special cases in the future
+	MinVNID = uint32(10)
+	// VNID: 0 reserved for default namespace and can reach any network in the cluster
+	GlobalVNID = uint32(0)
+
+	// ChangePodNetworkAnnotation is an annotation on NetNamespace to request change of pod network
+	ChangePodNetworkAnnotation string = "pod.network.openshift.io/multitenant.change-network"
+
+	// Acceptable values for ChangePodNetworkAnnotation
+	GlobalPodNetwork  PodNetworkAction = "global"
+	JoinPodNetwork    PodNetworkAction = "join"
+	IsolatePodNetwork PodNetworkAction = "isolate"
+)
+
+var (
+	ErrorPodNetworkAnnotationNotFound = fmt.Errorf("ChangePodNetworkAnnotation not found")
+)
+
+// Check if the given vnid is valid or not
+func ValidVNID(vnid uint32) error {
+	if vnid == GlobalVNID {
+		return nil
+	}
+	if vnid < MinVNID {
+		return fmt.Errorf("VNID must be greater than or equal to %d", MinVNID)
+	}
+	if vnid > MaxVNID {
+		return fmt.Errorf("VNID must be less than or equal to %d", MaxVNID)
+	}
+	return nil
+}
+
+// GetChangePodNetworkAnnotation fetches network change intent from NetNamespace
+func GetChangePodNetworkAnnotation(netns *NetNamespace) (PodNetworkAction, string, error) {
+	if netns.Annotations == nil {
+		return PodNetworkAction(""), "", ErrorPodNetworkAnnotationNotFound
+	}
+	value, ok := netns.Annotations[ChangePodNetworkAnnotation]
+	if !ok {
+		return PodNetworkAction(""), "", ErrorPodNetworkAnnotationNotFound
+	}
```

Comment:
you don't need the nil check; if it's nil, then "value, ok := ..." will return ok=false